BinaryEncoding.md

```diff
@@ -63,17 +57,14 @@ sequence, then followed recursively by any child nodes.
 
 * Examples
   * Given a simple AST node: `I32Add(left: AstNode, right: AstNode)`
-    * First write the opcode for `I32Add` (uint8)
+    * First write the opcode for `I32Add` (`varuint32`)
```

Comment:
Yeah, I had been thinking along these lines too (as a future binary iteration); I didn't realize the switch to LEB128 would be such a pain for V8 as an intermediate step.

On the subject: the opcode table would mean that the actual choice of indices was up to the module.  I had been thinking that the way to achieve what you're saying (255 ops fit in one byte instead of just 128) is just to use a variable-length scheme where 0xff was the two-byte prefix (and 0xffff mean 3 bytes, etc).  That's what I'll probably do in SM right now to workaround the fact that, for our internal asm.js->wasm conversion, we have >256 opcodes due to SIMD.